osu.Game/Rulesets/Ruleset.cs

```diff
@@ -19,8 +19,21 @@ public abstract class Ruleset
 
         public virtual IEnumerable<BeatmapStatistic> GetBeatmapStatistics(WorkingBeatmap beatmap) => new BeatmapStatistic[] { };
 
+        public abstract IEnumerable<Mod> GetAllMods();

```

Comment:
this currently leads to each rulesest having to duplicate it's mods for both `GetAllMods` and `GetModsFor`